File: app/schemas/admin_audit.py
"""Admin audit log schemas."""

from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class AdminAuditLogResponse(BaseModel):
    id: UUID
    actor_user_id: UUID
    action: str
    resource_type: str
    resource_id: str | None
    summary: str
    metadata_json: dict = Field(default_factory=dict)
    created_at: datetime

    model_config = {"from_attributes": True}


class AdminAuditLogListResponse(BaseModel):
    items: list[AdminAuditLogResponse]
    total: int
    page: int
    page_size: int

File: front/shared/types/admin-audit.ts
export interface AdminAuditLogResponse {
  id: string
  actor_user_id: string
  action: string
  resource_type: string
  resource_id?: string | null
  summary: string
  metadata_json: Record<string, unknown>
  created_at: string
}

export interface AdminAuditLogListResponse {
  items: AdminAuditLogResponse[]
  total: number
  page: number
  page_size: number
}
